### ISSUE TITLE
Failed to launch EndpointManager

### ISSUE THREAD

**Issue by NhatLeTTCU** [microsoft/mindaro#189]
I am using Windows and I am not able to run Bridge to Kubernetes in Visual Studio 2019. It has worked perfectly before. 

![Error](https://user-images.githubusercontent.com/74934383/122122456-709c4000-cdf2-11eb-8fe0-228c50d415a6.png)
[logfile.log](https://github.com/microsoft/mindaro/files/6658688/logfile.log)

**Comment by daniv-msft** [comment]
(Adding @amsoedal who is looking at customer issues this week)

Thanks @NhatLeTTCU for reporting this issue, and sorry you're getting this error. To help us investigate further, could you please attach the latest logs from your `%TEMP%/Bridge to Kubernetes` folder?
Given the error you're seeing, the latest logs called `bridge-library-...` and `bridge-endpointmanager-...` should be the most interesting ones.

If you don't want to attach the logs to this public Github issue, you can also send them to us by email at BridgeToKubernetes@microsoft.com.

**Comment by NhatLeTTCU** [comment]
@daniv-msft  Thanks for your response. I already attached logfile.log under the image. Please let me know if you already get that. Thanks.

**Comment by daniv-msft** [comment]
Thanks @NhatLeTTCU. I did see that, but unfortunately this log file only include a part of the information (the Visual Studio logs). The other logs mentioned above will include more information to help us investigate.

**Comment by NhatLeTTCU** [comment]
@daniv-msft oh, I got it. Already sent an email to your team. Thanks.

**Comment by amsoedal** [comment]
Hi @NhatLeTTCU, is this reproducing consistently or only sometimes?

**Comment by NhatLeTTCU** [comment]
@amsoedal It happens consistently. I do not see any EndpointManager under /.bridge, in that .bridge folder, there are 2 files only (config.json and config.json.bak)
Thanks.

**Comment by amsoedal** [comment]
@NhatLeTTCU OK thanks. Did this start happening after an upgrade? If so, can you try uninstalling and re-installing the extension? If not, I can follow up with some further steps on email.

**Comment by NhatLeTTCU** [comment]
@amsoedal No, I have not upgraded the extension. I did try uninstalling and re-installing also. It does not work.

**Comment by amsoedal** [comment]
@NhatLeTTCU thanks for giving that a go. I'm typing up some instructions to send to you over email at the moment. If you could please try one more thing: could you run `dotnet --list-runtimes`, and `dotnet --list-sdks`? I want to see if you have dotnet installed, and which version if so.

**Comment by amsoedal** [comment]
Update: The EndpointManager was failing to come up because the dotnet 3.1 runtime wasn't present. We have a fix in the works for this limitation, but if anyone else runs into this issue, please do install the dotnet runtime [here](https://dotnet.microsoft.com/download/dotnet/3.1) and try again.

**Comment by sunshine69** [comment]
I dont understand the dependencies on dotnet 3.1, pls make it clear, that the vscode bridge to k8s depends on a local dotnet runtime 3.1?

I thought it is not the case but I might be wrong.

In one of my machine I have the similar symptom like this ticket, on the other one I don't have this issues, but another. Will log in a different ticket now...

**Comment by sunshine69** [comment]
OK I got the same problems in the ubuntu 21 machine with exactly the same version of vscode and plugin. However I don't have problem in ubuntu 20.04.

The log does not have much usefull output except this
```
2021-06-19T23:15:34.133Z | Common Extension Root    | ERROR | Error: cli-client-connect-error <json>{}</json> <stack>Error: An unexpected error occurred: 'Failed to launch EndpointManager.'\nTo see our active issues or file a bug report, please visit https://aka.ms/bridge-to-k8s-report.\nFor diagnostic information, see logs at '/tmp/Bridge To Kubernetes'.\n\n   at ChildProcess.<anonymous> (/home/stevek/.vscode/extensions/mindaro.mindaro-1.0.120210615/dist/extension.js:3:143256)\n    at ChildProcess.emit (events.js:315:20)\n   at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)</stack>
2021-06-19T23:15:34.135Z | Connect (bridge-k8s-test | ERROR | Error: connect-error <json>{}</json> <stack>Error: An unexpected error occurred: 'Failed to launch EndpointManager.'\nTo see our active issues or file a bug report, please visit https://aka.ms/bridge-to-k8s-report.\nFor diagnostic information, see logs at '/tmp/Bridge To Kubernetes'.\n\n  at ChildProcess.<anonymous> (/home/stevek/.vscode/extensions/mindaro.mindaro-1.0.120210615/dist/extension.js:3:143256)\n    at ChildProcess.emit (events.js:315:20)\n   at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)</stack>
2021-06-19T23:15:34.137Z | Common Extension Root    | ERROR | Error: connect-service-task-terminal-error <json>{}</json> <stack>Error: Failed to establish a connection. Error: An unexpected error occurred: 'Failed to launch EndpointManager.'\nTo see our active issues or file a bug report, please visit https://aka.ms/bridge-to-k8s-report.\nFor diagnostic information, see logs at '/tmp/Bridge To Kubernetes'.\n\n   at /home/stevek/.vscode/extensions/mindaro.mindaro-1.0.120210615/dist/extension.js:3:77473\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)</stack>
```

**Comment by sunshine69** [comment]
Nevertheless I tried to install the dotnet asp runtime aspnetcore-runtime-3.1.16-linux-x64.tar.gz but problem remains.

For the working ubuntu 20.04 I do not need to have anything.

This is a hidden dependencies and poor error logging. Please fix it and let us know what dependencies is ...
Thanks

**Comment by sunshine69** [comment]
I install a fresh ubutntu 20.04 and start from the scratch. It turns out that the problem exists no matter what, I even did install donet sdk 3.1 as well without success.

My old existing ubuntu 20.04 woks but I am not sure what makes it works.

So the bugs does still exits and prevent me from using the plugin, could someone please have a look urgently? We are investigating the feasibility at this stage and it seems to be a blocker for us.

Thanks

**Comment by amsoedal** [comment]
@sunshine69 sorry that you're reproducing this consistently! Could you please attach the logs from /tmp/Bridge to Kubernetes (the most recent bridge-library, bridge-endpointmanager, and bridge-mindarocli logs)? You can also email them to us at BridgeToKubernetes@microsoft.com. It might be a separate issue from the one that Nhat is reproducing here, since we have seen Linux-specific issues in the past.

**Comment by sunshine69** [comment]
Another issues gave me the solution. Apparently for my case I have to do

```
export DOTNET_ROOT=/path/to/donet/rootdir
export BRIDGE_BINARYUTILITYVERSION=v1
```

**Comment by amsoedal** [comment]
@sunshine69 ok, thank you for clarifying! Glad you got unblocked.
